### PR TITLE
bump page limit in fetch_latest_commit_sha, spit out more meaningful error if too many branches were encountered

### PR DESCRIPTION
**Pull request by boegel**
too many branches may break `fetch_latest_commit_sha`, see e.g. https://travis-ci.org/hpcugent/easybuild-framework/jobs/174153835

### CODE REVIEW

File: easybuild/tools/github.py [hunk 1]
```diff
@@ -264,7 +264,10 @@ def fetch_latest_commit_sha(repo, account, branch='master', github_user=None, to
             break
 
     if res is None:
-        raise EasyBuildError("No branch with name %s found in repo %s/%s (%s)", branch, account, repo, data)
+        error_msg = "No branch with name %s found in repo %s/%s" % (branch, account, repo)
+        if len(data) >= GITHUB_MAX_PER_PAGE:
```
Comment (wpoely86) [inline easybuild/tools/github.py]: how does this work? Github will only return what you requested?

Comment (boegel) [inline easybuild/tools/github.py]: well, there shouldn't be more than `GITHUB_MAX_PER_PAGE` hits, indeed, so I could also check with `==`, but it doesn't matter...

Comment (boegel) [inline easybuild/tools/github.py]: bleh, it's far from trivial to do this, since the header info isn't readily available in the result of `github_api_get_request` (which is used by `fetch_latest_commit_sha`)...

Is it worth the trouble? This is only about a slightly more informative error message...

Comment (boegel) [inline easybuild/tools/github.py]: OK, but that should be tackled in a separate PR, it shouldn't block this one.

I opened an issue for that, see #1988

**Comment by boegel** [comment]
This is good to go, follow-up in #1988, thanks for the review @wpoely86!
